arch/risc-v/src/common/riscv_fpucmp.c

```diff
@@ -37,58 +34,27 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* Configuration ************************************************************/
-
-#undef HAVE_FPU
-#if defined(CONFIG_ARCH_FPU) && \
-    !defined(CONFIG_TESTING_OSTEST_FPUTESTDISABLE) && \
-    defined(CONFIG_TESTING_OSTEST_FPUSIZE) && \
-    defined(CONFIG_SCHED_WAITPID)
-#    define HAVE_FPU 1
-#endif
-
-#ifdef HAVE_FPU
-
-#if CONFIG_TESTING_OSTEST_FPUSIZE != (4 * FPU_XCPT_REGS)
-#  error "CONFIG_TESTING_OSTEST_FPUSIZE has the wrong size"
-#endif
+#ifdef CONFIG_ARCH_FPU
 
 /****************************************************************************
- * Private Data
+ * Public Functions
  ****************************************************************************/
 
-static uint32_t g_saveregs[XCPTCONTEXT_REGS];
-
 /****************************************************************************
- * Public Functions
+ * Name: up_fpucmp
+ *
+ * Description:
+ *   compare FPU areas from thread context
+ *
  ****************************************************************************/
 
-/* Given an array of size CONFIG_TESTING_OSTEST_FPUSIZE, this function will
- * return the current FPU registers.
- */
-
-void arch_getfpu(FAR uint32_t *fpusave)
+bool up_fpucmp(FAR const void *saveregs1, FAR const void *saveregs2)
```

Comment:
Let's remove `FAR` as it is not available for risc-v

Reply:
Done